### DISCUSSION TITLE
Help needed in understanding routing 2i2 4th gen

### DISCUSSION THREAD

**Discussion by edzob** [geoffreybennett/alsa-scarlett-gui#110] [General]
Hi, I am new to the focusrite scarlet domain. 

Just installed the alsa-scarlett-gui on arch linux and installed new firmware via scarlett2.
My goal is to change the routing in my scarlett, so that I when I am in a call (Signal, Google Meet, etc), 
that I can hear the attendee, and I can hear for example music via spotify-web,
and that they can only hear my voice via the microphone.

At this moment the default setup seems to be that the attendee also hear the music from other chrome tabs.

How can I change this?
I looked at https://github.com/geoffreybennett/alsa-scarlett-gui/blob/master/docs/iface-4th-gen.md
but I think I miss a bit of context to understand what I am reading.

My guess is that I need to remove a PCM input to DSP output routing?

Thanks for guiding me on my discovery journey. 

Regards

**Comment by geoffreybennett** [comment]
Hi @edzob,

It's more likely that you need to configure your sound server to not mix audio from all input channels. It's probably including loopback. Try the Pro Audio config in pavucontrol if you're using Pipewire. Otherwise, please answer the questions from the [New Issue Template](https://github.com/geoffreybennett/alsa-scarlett-gui/issues/new?template=issue.md).

Regards,
Geoffrey.

**Comment by edzob** [comment]
Hi @geoffreybennett  thanks for helping so quickly
I also have used for many years the Zoom H6n as usb xlr interface, also with the headphone connected to it,
and there the microphone was the only input when I selected the microphone in the gnome-sound setting.
That is why I thought it would be a device specific setting.

With the sound server you are pointing to something at the Arch linux side? 
I will google for Pro Audio pavucontrol to find out if there is a specific setting for the scarlet USB device in comparison to the Zoom USB device. Thanks for helping me finding out how this all links to each other.

Regards

**Comment by geoffreybennett** [comment]
Hi @edzob,

Your sound server would be PulseAudio, JACK, or PipeWire. Please see the New Issue Template linked above and paste the output from those commands here.

Don't disconnect the DSP Output → PCM Input routing ("(2)" in the diagram below) or you will have no microphone input!

![image](https://github.com/geoffreybennett/alsa-scarlett-gui/assets/1866657/0c46a106-1e3d-4682-bc74-33405a377239)

You could disconnect the connections marked (5) in the diagram to "fix" the problem, but then the microphone input will likely be too soft because the unused silent channels will still be mixed in. Better to fix the issue in the correct place.

Please do this:

- Install `pavucontrol`, run it, and select the Pro Audio configuration. That will likely just fix things by making recordings by default use the first two input channels.

I am working on better solution which uses UCM2 device profiles which will split the device into multiple virtual devices. If you can try it out and give feedback, that'd be great! The steps to install:

1) Download this ZIP file: [Scarlett-2i2-4th-Gen.zip](https://github.com/geoffreybennett/alsa-scarlett-gui/files/14904474/Scarlett-2i2-4th-Gen.zip)

2) That ZIP file has two conf files in it. Place those two files in the `/usr/share/alsa/ucm2/USB-Audio/Focusrite` directory.

3) Edit the `/usr/share/alsa/ucm2/USB-Audio/USB-Audio.conf` file (make a backup of it first). Search for `If.focusrite-scarlett-2i`. You will see many sections looking a lot like this one, which you should copy/paste in right after the existing Focusrite section:

```
If.focusrite-scarlett-2i2-4th-gen {
        Condition {
                Type RegexMatch
                String "${CardComponents}"
                Regex "USB1235:8219"
        }
        True.Define {
                ProfileName "Focusrite/Scarlett-2i2-4th-Gen"
        }
}
```

The new bit (highlighted below) should look like this in context:
![image](https://github.com/geoffreybennett/alsa-scarlett-gui/assets/1866657/59048b50-253f-4a15-a41e-594778bdc33f)

After that, restart your sound server (or if you don't know how, just reboot). Then in your recording software or the gnome sound settings (or in pavucontrol which I've found often works better than gnome sound settings) you should be able to select one of:
- Mic/Line/Inst In 1 (mono recording, first mic/line input)
- Mic/Line/Inst In 2 (mono recording, second mic/line input)
- Mic/Line/Inst In 1-2 (stereo recording, first and second mic/line inputs)
- Loopback (stereo recording, audio from your computer)

Let me know how you go! Any issues, please send screenshots and copies of the files that you edited.

Regards,
Geoffrey.

**Comment by edzob** [comment]
> Better to fix the issue in the correct place.

Yes! 
Thanks again for the elaborate help. and amazing that you know all this.
I will come back when I followed all the steps and tried the issue template.
Learning a lot by this :) aweomse.

**Comment by geoffreybennett** [comment]
>  Thanks again for the elaborate help. and amazing that you know all this.

No problem, I'm just learning too and sharing what I find :)

**Comment by edzob** [comment]
I seem not to have the "Pro Audio"  option in the `pavucontrol`.
My assumption, based on the checklist below, is that I have pipewire installed, but not as soundserver.
Since the [pipewire wiki](https://gitlab.freedesktop.org/pipewire/pipewire) stated that I should see `Server Name: PulseAudio (on PipeWire 0.3.x)`

I will google that (and read first the rest of your post content) :)

Here is the checklist from the template:
## Environment Details

### Linux Distribution and Version

```
$ cat /etc/lsb-release 
DISTRIB_ID="Arch"
DISTRIB_RELEASE="rolling"
DISTRIB_DESCRIPTION="Arch Linux"
```

### Kernel Version

```
$ uname -r
6.8.4-arch1-1
```

### Kernel Messages

```
$ sudo dmesg | grep -A 5 -B 5 -i focusrite
[   25.517279] hid-generic 0003:0FD9:0066.0005: hiddev98,hidraw4: USB HID v1.01 Device [Elgato Cam Link 4K] on usb-0000:00:14.0-1.4.4.2/input2
[   25.604122] usb 1-3.4.1: new high-speed USB device number 11 using xhci_hcd
[   25.744441] usb 1-3.4.1: New USB device found, idVendor=1235, idProduct=8219, bcdDevice= 8.50
[   25.744446] usb 1-3.4.1: New USB device strings: Mfr=1, Product=3, SerialNumber=2
[   25.744448] usb 1-3.4.1: Product: Scarlett 2i2 4th Gen
[   25.744449] usb 1-3.4.1: Manufacturer: Focusrite
[   25.744450] usb 1-3.4.1: SerialNumber: S2BHQDQ3B1EC1C
[   25.804216] usb usb4-port1: attempt power cycle
[   25.867471] usb 1-3.4.4: new high-speed USB device number 12 using xhci_hcd
[   26.014752] usb 1-3.4.4: New USB device found, idVendor=17ef, idProduct=30b9, bcdDevice=55.63
[   26.014762] usb 1-3.4.4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
--
[   27.824154] usb 1-3.4.4.4: New USB device found, idVendor=17ef, idProduct=30bb, bcdDevice= 0.41
[   27.824166] usb 1-3.4.4.4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[   27.824172] usb 1-3.4.4.4: Product: ThinkPad Thunderbolt 4 Dock USB Audio
[   27.824176] usb 1-3.4.4.4: Manufacturer: Lenovo
[   27.824180] usb 1-3.4.4.4: SerialNumber: 000000000000
[   27.836152] usb 1-3.4.1: Focusrite Scarlett Gen 4 Mixer Driver enabled (pid=0x8219); report any issues to https://github.com/geoffreybennett/scarlett-gen2/issues
[   27.836416] usb 1-3.4.1: Firmware version 2128
[   27.877094] hid-generic 0003:17EF:30BB.0006: hiddev99,hidraw5: USB HID v1.11 Device [Lenovo ThinkPad Thunderbolt 4 Dock USB Audio] on usb-0000:00:14.0-3.4.4.4/input3
[   27.877264] usbcore: registered new interface driver snd-usb-audio
[   28.177872] usbcore: registered new device driver r8152-cfgselector
[   28.257671] r8152-cfgselector 2-1.4.4.3: reset SuperSpeed USB device number 6 using xhci_hcd
```

### Focusrite Interface Series and Model

```
$ lsusb -d1235:
Bus 001 Device 011: ID 1235:8219 Focusrite-Novation Scarlett 2i2 4th Gen
```

```
$ scarlett2 -l
Found 1 supported device:
 card2: Scarlett 4th Gen 2i2 (firmware version 2128)
```

### Audio System

```
$ ps aux | grep -E "pulseaudio|jackd|pipewire"
edzob       3531  0.0  0.0 106556 10036 ?        S<sl 09:17   0:00 /usr/bin/pipewire
edzob       3532  0.0  0.0  92740  6236 ?        Ssl  09:17   0:00 /usr/bin/pipewire-media-session
edzob       3533  1.7  0.0 2131980 30884 ?       S<sl 09:17   1:07 /usr/bin/pulseaudio --daemonize=no --log-target=journal
edzob      37046  0.0  0.0   6976  4272 pts/0    S+   10:23   0:00 grep --color=auto -E pulseaudio|jackd|pipewire
```

```
$ pactl info
Server String: /run/user/1000/pulse/native
Library Protocol Version: 35
Server Protocol Version: 35
Is Local: yes
Client Index: 24
Tile Size: 65472
User Name: edzob
Host Name: dell202305
Server Name: pulseaudio
Server Version: 17.0
Default Sample Specification: s16le 2ch 44100Hz
Default Channel Map: front-left,front-right
Default Sink: alsa_output.usb-Focusrite_Scarlett_2i2_4th_Gen_S2BHQDQ3B1EC1C-00.iec958-stereo
Default Source: alsa_input.usb-Focusrite_Scarlett_2i2_4th_Gen_S2BHQDQ3B1EC1C-00.multichannel-input
Cookie: f24d:6ce0
```

**Comment by edzob** [comment]
A small addition for the knowledge base. This fixed the pactl info message that pipewire was not active as soundserver, and now I have the option of `Pro Audio` in the setting of the `Scarlett` in `pavucontrol`

Nice page on how to install pipewire in Arch 
[LinuxGinie: How to Install PipeWire on Arch Linux?](https://linuxgenie.net/install-pipewire-on-arch-linux/)

I double checked the content with [Arch Wiki: PipeWire](https://wiki.archlinux.org/title/PipeWire)
